apps/jobs/views.py

```diff
@@ -393,112 +401,47 @@ def leaderboard(request, challenge_phase_split_id):
-        }
-        return Response(response_data, status=status.HTTP_200_OK)
 
+@throttle_classes([UserRateThrottle])
+@api_view(['GET'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_remaining_submissions_for_all_phases(request, challenge_pk):
+    phases_data = {}
+    try:
+        challenge = Challenge.objects.get(pk=challenge_pk)
+    except Challenge.DoesNotExist:
+        response_data = {'error': 'Challenge does not exist'}
+        return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+    if is_user_a_host_of_challenge(request.user, challenge_pk):
+        challenge_phase = ChallengePhase.objects.filter(
+            challenge=challenge).order_by('pk')
     else:
-        # calculate the remaining submissions from total submissions.
-        remaining_submission_count = max_submissions_count - submissions_done_count
-        # Calculate the remaining submissions for current month.
-        remaining_submissions_this_month_count = (max_submissions_per_month_count -
-                                                  submissions_done_this_month_count)
-        # Calculate the remaining submissions for today.
-        remaining_submissions_today_count = (max_submissions_per_day_count -
-                                             submissions_done_today_count)
-
-        remaining_submissions_this_month_count = min(remaining_submission_count,
-                                                     remaining_submissions_this_month_count)
-        remaining_submissions_today_count = min(remaining_submissions_this_month_count,
-                                                remaining_submissions_today_count)
-
-        response_data = {
-            'remaining_submissions_this_month_count': remaining_submissions_this_month_count,
-            'remaining_submissions_today_count': remaining_submissions_today_count,
-            'remaining_submissions': remaining_submission_count
-        }
-        return Response(response_data, status=status.HTTP_200_OK)
+        challenge_phase = ChallengePhase.objects.filter(
+            challenge=challenge, is_public=True).order_by('pk')
+    phases_data['participant_team'] = get_participant_team_name_of_user_for_a_challenge(
+            request.user, challenge_pk)
+    phases_data['participant_team_id'] = get_participant_team_id_of_user_for_a_challenge(
+            request.user, challenge_pk)
+    phase_data_list = list()
+    for phase in challenge_phase:
+        remaining_submission_message, response_status = get_remaining_submission_for_phase(request.user,
+                                                                                           phase.id,
+                                                                                           challenge_pk)
+        if response_status != status.HTTP_200_OK:
+            return Response(remaining_submission_message, status=response_status)
```

Comment:
Why do we need this condition? Am I missing something here?

Reply:
This is there to handle 403 from [`Conditional check for the existence of participant team of the user`](https://github.com/Cloud-CV/EvalAI/blob/c237a07fda872cff5b79975898544ca6250dbae9/apps/jobs/utils.py#L27) or some other checks added later.